syntax = "proto3";

package user;

service User {
  rpc SignUp (SignUpRequest) returns (UserResponse);
  rpc EditProfile (EditProfileRequest) returns (UserResponse);
  rpc Login (LoginRequest) returns (LoginResponse);
}

message SignUpRequest {
  string email = 1;
  string username = 2;
  string name = 3;
  string department = 4;
  string password = 5;
}

message EditProfileRequest {
  int32 user_id = 1;
  string email = 2;
  string username = 3;
  string name = 4;
  string department = 5;
}

message LoginRequest {
  string username = 1;
  string password = 2;
}

message UserResponse {
  int32 id = 1;
  string email = 2;
  string username = 3;
  string name = 4;
  string department = 5;
}

message LoginResponse {
  string token = 1;
  UserResponse user = 2;
}
